tests/expecter.go

```diff
@@ -0,0 +1,402 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package tests
+
+import (
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+
+	expect "github.com/google/goexpect"
+	"google.golang.org/grpc/codes"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/client-go/log"
+	"kubevirt.io/kubevirt/pkg/util/net/dns"
+)
+
+const (
+	PromptExpression = `(\$ |\# )`
+	CRLF             = "\r\n"
+)
+
+var (
+	shellSuccess = regexp.MustCompile(RetValue("0"))
+	shellFail    = regexp.MustCompile(RetValue("[1-9].*"))
+)
+
+// VmiConsoleExpectBatch runs the batch from `expected` connecting to an already logged in console at vmi and
```

Comment:
It does not expect anything to be logged in, one can use it to perform the login as well.
Maybe the caller expects it to be logged in already based on how he composes the batch.

Reply:
Done.